binary-string-types.ipynb

```diff
@@ -0,0 +1,530 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Variable sized types\n",
+    "\n",
+    "## Variable length binary, large binary, string and large string\n",
+    "\n",
+    "Binary and string types share the same physical layout with where the string type is utf-8 binary and will produce an invalid result if the bytes are not valid utf-8.\n",
+    "\n",
+    "The difference the between binary/string and large binary/string is in the offset type. In the first case that is `int32` and in the second it is `int64`. Second in generally used for data larger than 2GB.\n",
+    "\n",
+    "The bytes of a binary or string column are stored together consecutively in a single buffer or region of memory. To know where each element of the column starts and ends the physical layout also includes integer offsets. The length of which is one more then the length on the column as the last two elements define the start and the end of the last element in the binary/string column.\n",
```

Comment:
I would swap the order of those two paragraphs, given that in the first you explain the difference between string and large string is the type of the offsets, but only in the second paragraph you introduce the concept of offsets